src/components/ChannelPreview/hooks/useChannelPreviewDisplayName.js

```diff
@@ -3,23 +3,29 @@ import { ChatContext } from '../../../context';
 
 export const useChannelPreviewDisplayName = (channel) => {
   const { client } = useContext(ChatContext);
-  const [displayName, setDisplayName] = useState('');
+  const [displayName, setDisplayName] = useState(
+    getChannelPreviewDisplayName(channel, client.user.id),
+  );
 
   useEffect(() => {
-    if (typeof channel?.data?.name === 'string') {
-      setDisplayName(channel.data.name);
-    } else {
-      const members = Object.values(channel?.state?.members || {});
-      const otherMembers = members.filter(
-        (member) => member.user.id !== client.user.id,
-      );
-      const name = otherMembers
-        .map((member) => member.user.name || member.user.id || 'Unnamed User')
-        .join(', ');
-
-      setDisplayName(name);
-    }
+    setDisplayName(getChannelPreviewDisplayName(channel, client.user.id));
   }, [channel]);
 
   return displayName;
 };
+
+const getChannelPreviewDisplayName = (channel, currentUserId) => {
+  if (typeof channel?.data?.name === 'string') {
+    return channel.data.name;
+  } else {
+    const members = Object.values(channel?.state?.members || {});
```

Comment:
in the case that members doesn't exist, do we want to return a default value for name instead of filter/map on an empty array?

Reply:
Lets leave it at empty name for now!! Will need to discuss with Josh about it